From c189729809e4c7a6298126a76db608da2b571240 Mon Sep 17 00:00:00 2001
From: Pu Lehui <pulehui@huawei.com>
Date: Wed, 18 Dec 2024 06:24:00 +0000
Subject: [PATCH 08/17] bpf: Add kabi reserve padding for uapi struct
 bpf_link_info

hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/IBC248

--------------------------------

Add kabi reserve padding for uapi struct bpf_link_info

Signed-off-by: Pu Lehui <pulehui@huawei.com>
---
 include/uapi/linux/bpf.h       | 9 +++++++++
 tools/include/uapi/linux/bpf.h | 9 +++++++++
 2 files changed, 18 insertions(+)

diff --git a/include/uapi/linux/bpf.h b/include/uapi/linux/bpf.h
index 482647774bf5..a660cb68c853 100644
--- a/include/uapi/linux/bpf.h
+++ b/include/uapi/linux/bpf.h
@@ -6573,6 +6573,15 @@ struct bpf_link_info {
 					__u64 config;
 					__u32 type;
 				} event; /* BPF_PERF_EVENT_EVENT */
+				struct {
+					__u64:64;
+					__u32:32;
+					__u32:32;
+					__u64:64;
+					__u64:64;
+					__u64:64;
+					__u64:64;
+				} kabi_reserve;
 			};
 		} perf_event;
 		struct {
diff --git a/tools/include/uapi/linux/bpf.h b/tools/include/uapi/linux/bpf.h
index c112c6f7c766..9b302242be6c 100644
--- a/tools/include/uapi/linux/bpf.h
+++ b/tools/include/uapi/linux/bpf.h
@@ -6576,6 +6576,15 @@ struct bpf_link_info {
 					__u64 config;
 					__u32 type;
 				} event; /* BPF_PERF_EVENT_EVENT */
+				struct {
+					__u64:64;
+					__u32:32;
+					__u32:32;
+					__u64:64;
+					__u64:64;
+					__u64:64;
+					__u64:64;
+				} kabi_reserve;
 			};
 		} perf_event;
 		struct {
-- 
2.25.1

